### PR TITLE
Add support for kind filter to RepoStore.List

### PR DESCRIPTION
**Pull request by asdine**

### CODE REVIEW

File: internal/db/repos.go [hunk 1]
```diff
@@ -408,6 +408,9 @@ type ReposListOptions struct {
 	// and this may be replaced by the version context name.
 	Names []string
 
+	// Kinds of repos to list. When zero-valued, this is omitted from the predicate set.
+	Kinds []string
```
Comment (ryanslade) [inline internal/db/repos.go:412]: I think you should be using the types, not kinds. I know, it's super confusing:

https://github.com/sourcegraph/sourcegraph/blob/c1bed31aa286e4d3291677c3cf67dd412686ff5e/internal/extsvc/types.go#L87

Comment (ryanslade) [reply internal/db/repos.go:412]: Or, you could ask for kinds and use extsvc.KindToType, then you don't need to care about casing

Comment (ryanslade) [reply internal/db/repos.go:412]: I think this is some debt we should work on

Comment (asdine) [reply internal/db/repos.go:412]: Good point, we might as well just use them

Comment (asdine) [reply internal/db/repos.go]: I have renamed them to `ServiceTypes` instead of `Types` to be a bit more explicit

Comment (ryanslade) [reply internal/db/repos.go]: https://github.com/sourcegraph/sourcegraph/issues/15385

**Review by ryanslade — APPROVED** [review]

**Pull request merged**